### PR TITLE
Passage / version metadata improvements

### PR DESCRIPTION
**Pull request by jacobwegner**
What's in this PR:

- Integrates a new siblings widget:
image
- Expands metadata:
image
- Adds the version label to the top of the reader pane:
image

Preview via:

https://feature-passage-metadata-improvements--explorehomer-dev.netlify.app/reader?urn=urn%3Acts%3AgreekLit%3Atlg0012.tlg001.msA-folios%3A12r

Depends on:
- [scaife-viewer/scaife-widgets#32](https://github.com/scaife-viewer/scaife-widgets/pull/32)
- [scaife-viewer/explorehomer-atlas#55](https://github.com/scaife-viewer/explorehomer-atlas/pull/55)

### CODE REVIEW

**Comment by jacobwegner** [comment]
[Display a human readable reference for the current passage](https://trello.com/c/IhfObzBy/174-display-a-human-readable-reference-for-the-current-passage)

**Comment by jtauber** [comment]
I wonder if we should make the references in the "relationship" widgets a smaller font size, drop the border (except maybe on hover), change the "active" in sibling to be white-on-brand rather than just bold.

Screen Shot 2020-06-26 at 2 48 26 AM

Here are the diffs in the above mockup:

```css
.passage-siblings-widget [data-v-612dab5a] {
  /* border: 1px solid #e9ecef; */
  /* font-size: .8rem; */
  font-size: .7rem;
  padding: 0.1rem 0.3rem;
}

.active-sibling[data-v-612dab5a] {
  background: #b45141;
  color: #FFF;
}

.passage-siblings-widget[data-v-612dab5a] {
  /* grid-gap: .0825em; */
}
```

**Comment by jtauber** [comment]
Just an experiment for the Metadata:

Screen Shot 2020-06-26 at 3 05 19 AM

But the key ideas are:

* put label and value one same line (wrap value if needed, perhaps use grid)
* distinguish label and value more (here by colour and typeface)
* use fixed-width for URN (if possible to do on a per-property basis like this)

**Comment by jtauber** [comment]
I did notice when fixed-height, the width of the grid seems too narrow and there's a horizontal scroll bar:

(my styling from above is still applied but I don't think that's causing this)

Screen Shot 2020-06-26 at 3 11 05 AM

File: src/reader/widgets/ReaderWidget.vue [hunk 1]
```diff
@@ -57,6 +57,19 @@
...
       },
+      versionMetadata: {
+        immediate: true,
+        handler() {
+          if (!this.versionMetadata) {
+            this.setVersionMetadata();
+          }
+          // @@@ determine if we want this hooked
+          // further into scaife-skeleton
+          document.querySelector(
```
Comment (jacobwegner) [inline src/reader/widgets/ReaderWidget.vue:68]: @paltman thoughts on adding a Portal for setting the header?

https://github.com/scaife-viewer/scaife-skeleton/blob/5d9034b349ed33ebe293756e721369fefa9497c7/src/skeleton/main/MainLayout.vue#L5

Or we could have a reactive property in Vuex e.g. `mainWidgetTitle`.

**Comment by jacobwegner** [comment]
> I wonder if we should make the references in the "relationship" widgets a smaller font size, drop the border (except maybe on hover), change the "active" in sibling to be white-on-brand rather than just bold.
> 
> Screen Shot 2020-06-26 at 2 48 26 AM
> 
> Here are the diffs in the above mockup:
> 
> ```css
> .passage-siblings-widget [data-v-612dab5a] {
>   /* border: 1px solid #e9ecef; */
>   /* font-size: .8rem; */
>   font-size: .7rem;
>   padding: 0.1rem 0.3rem;
> }
> 
> .active-sibling[data-v-612dab5a] {
>   background: #b45141;
>   color: #FFF;
> }
> 
> .passage-siblings-widget[data-v-612dab5a] {
>   /* grid-gap: .0825em; */
> }
> ```

@jtauber I like your changes and will work on implementing them within EH.  I had wanted to do the brand background white color idea (overriding in EH) but I had also wanted to pick @paltman's brain to see if he had thoughts on how we'd be able to implement the styles within the widget proper, given that:

- `$brand` is defined at the project level
- the styles are defined at the widget level

@paltman: If we wanted the widget to make use of a variable for `$brand`, how would we best do that?

(we can certainly revisit this in our "post-scaife-widgets/skeleton" world, but I'm curious if there is prior art to follow from other Vue packages

**Comment by jacobwegner** [comment]
> Just an experiment for the Metadata:
> 
> Screen Shot 2020-06-26 at 3 05 19 AM
> 
> But the key ideas are:
> 
> * put label and value on same line (wrap value if needed, perhaps use grid)
> * distinguish label and value more (here by colour and typeface)

Yep; I'll do my best to implement.

> * use fixed-width for URN (if possible to do on a per-property basis like this)

Longer term, I want to change how metadata is queried from ATLAS and we will likely have more control.  Shorter term, we can hardcode it.

The other thing we could do is move the URN out of metadata and to its own widget:

image

**Comment by jacobwegner** [comment]
@jtauber I've updated the siblings widget and took a pass at the metadata widget.

I couldn't quite get flexbox or a grid to do what you did in your mockup; you're welcome to make suggestions or provide a suggestion on https://github.com/scaife-viewer/scaife-widgets/pull/32/files#r445874727

image

**Comment by jacobwegner** [comment]
image

image

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "@fortawesome/fontawesome-svg-core": "^1.2.28",
     "@fortawesome/free-solid-svg-icons": "^5.13.0",
     "@fortawesome/vue-fontawesome": "^0.1.9",
-    "@scaife-viewer/scaife-widgets": "0.13.1",
+    "@scaife-viewer/scaife-widgets": "git://github.com/scaife-viewer/scaife-widgets#6fd5e8b8e2ce7f650cec0720efe19bb307d7f285",
```
Comment (jacobwegner) [inline package.json:16]: - [ ] Bump to release of `scaife-widgets` after geting feedback on https://github.com/scaife-viewer/scaife-widgets/pull/32

File: src/App.vue [hunk 3]
```diff
@@ -39,4 +39,7 @@
   a {
     color: $explorehomer-brand;
   }
+  div.passage-siblings-widget > .grid-cell-square > .active-sibling {
+    background-color: $explorehomer-brand;
```
Comment (jacobwegner) [inline src/App.vue:43]: - [ ] Pass $brand in to scaife-widgets

File: src/reader/widgets/ReaderWidget.vue [hunk 4]
```diff
@@ -63,6 +63,21 @@
...
         });
       },
+      versionMetadata: {
+        immediate: true,
+        handler() {
+          if (!this.versionMetadata) {
+            this.setVersionMetadata();
+          }
+          else {
+            // @@@ determine if we want this hooked
```
Comment (jacobwegner) [inline src/reader/widgets/ReaderWidget.vue:73]: - [ ] Review for a better hook (possibly portal?)

File: src/reader/widgets/ReaderWidget.vue (hunk 1 above)
Comment (jacobwegner) [reply src/reader/widgets/ReaderWidget.vue]: @paltman suggest moving this down into `ReaderWidget`

File: src/App.vue (hunk 3 above)
Comment (jacobwegner) [reply src/App.vue]: resolved via 72d41f435c6f1c5331c6e30c40cf5309f6554e65

File: src/reader/widgets/ReaderWidget.vue (hunk 1 above)
Comment (jacobwegner) [reply src/reader/widgets/ReaderWidget.vue]: resolved via c8c47c6c0b46fe04bcf040c2fbaefc7e750418a3

**Comment by jacobwegner** [comment]
@paltman thanks; when you're able, please check out https://github.com/scaife-viewer/scaife-widgets/pull/32 as well, as that is required for the changes here

**Pull request merged**